packages/docmanager-extension/src/index.ts

```diff
@@ -943,4 +954,33 @@ namespace Private {
     body.appendChild(lastCheckpointMessage);
     return body;
   }
+
+  /**
+   * Ask user for a checkpoint to revert to.
+   */
+  export async function getTargetCheckpoint(
+    checkpoints: Contents.ICheckpointModel[],
+    trans: TranslationBundle
+  ): Promise<Contents.ICheckpointModel | undefined> {
+    // the id could be too long to show so use the index instead
+    const indexSeparator = '.';
+    const items = checkpoints.map((checkpoint, index) => {
+      const isoDate = Time.format(checkpoint.last_modified);
+      const humanDate = Time.formatHuman(checkpoint.last_modified);
+      return `${index}${indexSeparator} ${isoDate} (${humanDate})`;
+    });
+
+    const selectedItem = (
+      await InputDialog.getItem({
+        items: items,
+        title: trans.__('Choose a checkpoint')
+      })
+    ).value;
+
+    if (!selectedItem) {
+      return;
+    }
+    const selectedIndex = parseInt(selectedItem.split(indexSeparator, 1)[0]);
```

Comment:
I would not include the index as the last_modified time will be unique.

```suggestion
    const items = checkpoints.map((checkpoint, index) => {
      const isoDate = Time.format(checkpoint.last_modified);
      const humanDate = Time.formatHuman(checkpoint.last_modified);
      return `${isoDate} (${humanDate})`;
    });

    const selectedItem = (
      await InputDialog.getItem({
        items: items,
        title: trans.__('Choose a checkpoint')
      })
    ).value;

    if (!selectedItem) {
      return;
    }
    const selectedIndex = items.findIndex(item => item === selectedItem);
```

Reply:
`last_modified` is unique indeed, but `Time.format` has resolution of minutes. It seems perfectly plausible that user may have more than one checkpoint per minute (e.g. one ever 30 seconds), which is also demonstrated in the GIF in this PR. I would say we keep the index.

Reply:
Good point. Thanks for highlighting that. So let's keep the index.

Reply:
It would have helped if `InputDialog.getItem`  allowed to set the values of the `<option>` tags it creates, that way we could just use the checkpoint id as value.